Sources/SIL/SILParser.swift

```diff
@@ -610,7 +630,7 @@ class SILParser: Parser {
     func parseTypeName() throws -> String {
         let start = position
         // TODO(#14): Make name parsing more thorough.
-        let name = take(while: { $0.isLetter || $0.isNumber || $0 == "_" })
+        let name = take(while: { $0.isLetter || $0.isNumber || "_[]".contains($0) })
```

Comment:
When is this necessary? Let's add a test?

Reply:
Apparently SILGen might contain desugared types 😕

Reply:
Here you go:
```
%149 = apply %148<[Int], PartialRangeFrom<Int>>(%143, %146, %144)
```

Reply:
Hmmm now that I think about it, it might be better to normalize `[T]` to `Array<T>`